pkg/config/piped.go

```diff
@@ -594,19 +591,31 @@ type PipedImageWatcher struct {
 }
 
 // Validate checks if the duplicated repository setting exists.
-func (i *PipedImageWatcher) Validate() error {
-	repos := make(map[string]struct{})
-	for _, repo := range i.Repos {
-		if _, ok := repos[repo.RepoID]; ok {
-			return fmt.Errorf("duplicated repo id (%s) found in the imageWatcher directive", repo.RepoID)
+// And it populates default value if not set.
+func (p *PipedImageWatcher) Validate() error {
+	repos := make(map[string]struct{}, len(p.Repos))
+	for i := 0; i < len(p.Repos); i++ {
+		if _, ok := repos[p.Repos[i].RepoID]; ok {
+			return fmt.Errorf("duplicated repo id (%s) found in the imageWatcher directive", p.Repos[i].RepoID)
+		}
+		repos[p.Repos[i].RepoID] = struct{}{}
+
+		if p.Repos[i].PullInterval == 0 {
+			p.Repos[i].PullInterval = defaultImageWatcherPullInterval
 		}
-		repos[repo.RepoID] = struct{}{}
 	}
 	return nil
 }
 
 type PipedImageWatcherRepoTarget struct {
 	RepoID string `json:"repoId"`
+	// Interval to pull from git repository and pull images defined
+	// in image watcher file in the repo from image provider.
+	// Default is 5m.
+	PullInterval Duration `json:"pullInterval"`
```

Comment:
I feel we need a better name, how about `checkInterval`?

Reply:
Looks good, `pullInterval` sounds heavy operation.